src/core/libmaven/mzSample.cpp

```diff
@@ -1257,16 +1257,19 @@ EIC *mzSample::getEIC(float precursorMz, float collisionEnergy, float productMz,
 		}
 
 		int lastIndex = e->rt.size()-1;
+		//if rt is already present save the higher intensity for that rt
+		//this can happen when there are multiple product m/z for the same precursor
 		if (lastIndex != -1 && scan->rt == e->rt[lastIndex]) {
```

Comment:
Correct me wherever I am wrong. Here we are checking two things:
1. whether the vector is empty:  for this we can simply use the empty() function
2. if eic already has the rt:  for this we can use the std::find() function. This will also help us in getting the index

I would suggest using the standard functions.

Reply:
@rish9511 Thanks. I'll do that.

Reply:
@rish9511 done.